### DISCUSSION TITLE
Is there a way to hide the 'embossed' look of  submenu?

### DISCUSSION THREAD

**Discussion by santokalayil** [fltk-rs/fltk-rs#701] [Q&A]
Is there a way to change the 'embossed' look of  submenu to a flat one? I could use the set frame(flatbox) in the main menu.. but in submenu section, how that to be done?
![issue3](https://user-images.githubusercontent.com/49450970/120068171-f3a46300-c09c-11eb-816e-23f973c2fbad.png)
I was following the menu code found in the texteditor example in the repository to create the menu..
Is it possible to customize fully the menu items..?

**Comment by MoAlyousef** [comment]
Setting the frame type for the main menu sets it for all submenues (like trying a RoundedBox): 

![Clipboard01](https://user-images.githubusercontent.com/37966791/120068785-29405080-c08b-11eb-956d-d5afcfbd0968.jpg)

Otherwise to get full customization, you would need to implement your own menu widget (like in the custom_popup example).

**Reply by santokalayil** [reply]
Setting shape is working. but it did not change the embossed style. I have set frame to main menu.. but it did not change the embossed appearance of the submenu.. The following is the used code.

```
impl Menu {
    pub fn new(s: &app::Sender<Message>, x:i32, y:i32, w:i32, h:i32) -> Self {
        let mut menu = menu::SysMenuBar::default().with_pos(x, y).with_size(w, h);
        menu.set_frame(fltk::enums::FrameType::FlatBox);
        // menu.set_color(Color::Yellow);
        menu.set_color(menu.parent().unwrap().color());
        menu.set_text_size(menu.parent().unwrap().label_size());
        menu.set_text_color(Color::White);
        menu.global(); // Make a menu globally accessible from any window
        menu.set_down_frame(fltk::enums::FrameType::FlatBox);
        menu.add_emit(
            "&File/New...\t",
            Shortcut::Ctrl | 'n',
            menu::MenuFlag::Normal,
            *s,
            Message::New,
        );

        menu.add_emit(
            "&File/Open...\t",
            Shortcut::Ctrl | 'o',
            menu::MenuFlag::Normal,
            *s,
            Message::Open,
        );

        menu.add_emit(
            "&File/Save\t",
            Shortcut::Ctrl | 's',
            menu::MenuFlag::Normal,
            *s,
            Message::Save,
        );

        menu.add_emit(
            "&File/Save as...\t",
            Shortcut::Ctrl | 'w',
            menu::MenuFlag::Normal,
            *s,
            Message::SaveAs,
        );

        menu.add_emit(
            "&File/Print...\t",
            Shortcut::Ctrl | 'p',
            menu::MenuFlag::MenuDivider,
            *s,
            Message::Print,
        );

        menu.add_emit(
            "&File/Quit\t",
            Shortcut::Ctrl | 'q',
            menu::MenuFlag::Normal,
            *s,
            Message::Quit,
        );

        menu.add_emit(
            "&Edit/Cut\t",
            Shortcut::Ctrl | 'x',
            menu::MenuFlag::Normal,
            *s,
            Message::Cut,
        );

        menu.add_emit(
            "&Edit/Copy\t",
            Shortcut::Ctrl | 'c',
            menu::MenuFlag::Normal,
            *s,
            Message::Copy,
        );

        menu.add_emit(
            "&Edit/Paste\t",
            Shortcut::Ctrl | 'v',
            menu::MenuFlag::Normal,
            *s,
            Message::Paste,
        );

        menu.add_emit(
            "&Help/About\t",
            Shortcut::None,
            menu::MenuFlag::Normal,
            *s,
            Message::About,
        );

        if let Some(mut item) = menu.find_item("&File/Quit\t") {
            item.set_label_color(Color::from_u32(0xffff99));
        }

        Self { _menu: menu, x, y, w, h }
    }
    // pub fn x(&self) -> i32 {self.x}
    // pub fn y(&self) -> i32 {self.y}
    pub fn w(&self) -> i32 {self.w}
    // pub fn h(&self) -> i32 {self.h}

    pub fn set_color(&mut self, color:Color) {
        self._menu.set_color(color);
    }

    // pub fn get_parent_color(&self) -> Color {
    //     self._menu.parent().unwrap().color()
    // }

}

```

**Reply by MoAlyousef** [reply]
You might want to change the style of the app's windows:
```rust
    app::set_frame_type(FrameType::BorderBox);
```

**Reply by santokalayil** [reply]
it worked with menu.. but changed every other components' look into embossed type.. It is okay I go with this one for time being..

**Reply by MoAlyousef** [reply]
If you use `app::set_frame_type(FrameType::FlatBox);`, then all other instances of `some_widget.set_frame(FrameType::FlatBox);` can be removed. The idea is that app::set_frame_type uses swapping as to not lose the default (swapped out frame type).

**Reply by santokalayil** [reply]
The same I have tried  setting  fltk app is set to frame_type FlatBox. But the whole window is shown like an embosed window when
`app::set_frame_type(FrameType::FlatBox);` is used. Until that time, window was looking like flat box but now it is an embossed one.

**Reply by santokalayil** [reply]
Fixed the issue of `app::set_frame_type(FrameType::FlatBox)` issue in the new version which will be soon updated. Now this fix can be used by changing fltk version in toml to file to the following: 
```
[dependencies]
fltk = { version = "1.0.17", git = "https://github.com/fltk-rs/fltk-rs" }
```
The iissue is answered in [this discussion](https://github.com/fltk-rs/fltk-rs/discussions/711#discussioncomment-825722)
